src/daemon/daemon.cpp

```diff
@@ -1129,14 +1129,33 @@ InstanceSnapshotsMap map_snapshots_to_instances(const InstanceSnapshotPairs& ins
 }
 
 bool verify_snapshot_picks(const InstanceSelectionReport& report,
-                           const std::unordered_map<std::string, SnapshotPick>& snapshot_picks)
+                           const std::unordered_map<std::string, SnapshotPick>& snapshot_picks,
+                           bool purge)
 {
     auto any_snapshot = false;
+    std::vector<std::string> snapshots_of_deleted_instances{};
     for (const auto* selection : {&report.deleted_selection, &report.operative_selection})
+    {
         for (const auto& vm_it : *selection)
+        {
             if (auto pick_it = snapshot_picks.find(vm_it->first); pick_it != snapshot_picks.end())
+            {
                 for (const auto& snapshot_name : pick_it->second.pick)
-                    any_snapshot = true, vm_it->second->get_snapshot(snapshot_name); // throws if missing
+                {
+                    if (selection == &report.deleted_selection && (!pick_it->second.all_or_none || !purge))
+                        snapshots_of_deleted_instances.push_back(fmt::format("{}.{}", vm_it->first, snapshot_name));
+
+                    vm_it->second->get_snapshot(snapshot_name); // throws if missing
```

Comment:
I think the original line `any_snapshot = true, vm_it->second->get_snapshot(snapshot_name);` equals 
```
any_snapshot = true; 
vm_it->second->get_snapshot(snapshot_name);
```
Should it be the same order in order to keep the same behavior since get_snapshot has a chance to throw?

Reply:
You're right that this changes the order, but I don't think it matters much. That boolean is supposed to be returned, but it won't be if the function exits with an exception. The previous order was also pretty much arbitrary.